.github/guides/STYLE.md

```diff
@@ -399,6 +400,328 @@ turn_on(power_usage = 30) // Fine!
 ```
 
+## Macros
+
+Macros are, in essence, direct copy and pastes into the code. They are one of the few zero cost abstractions we have in DM, and you will see them often. Macros have strange syntax requirements, so if you see lots of backslashes and semicolons and braces that you wouldn't normally see, that is why.
+
+This section will assume you understand the following concepts:
+
+### Language - Hygienic
+We say a macro is [**hygienic**](https://en.wikipedia.org/wiki/Hygienic_macro) if, generally, it does not rely on input not given to it directly through the call site, and does not affect the call site outside of it in a way that could not be easily reused somewhere else.
+
+An example of a non-hygienic macro is:
+
+```dm
+#define GET_HEALTH(health_percent) (##health_percent * max_health)
+```
+
+In here, we rely on the external `max_health` value.
+
+Here are two examples of non-hygienic macros, because it affects its call site:
+
+```dm
+#define DECLARE_MOTH(name) var/mob/living/moth/moth = new(##name)
+#define RETURN_IF(condition) if (condition) { return; }
+```
+
+### Language - Side effects/Pure
+We say something has [**side effects**](https://en.wikipedia.org/wiki/Side_effect_(computer_science)) if it mutates anything outside of itself. We say something is **pure** if it does not.
+
+For example, this has no side effects, and is pure:
+```dm
+#define MOTH_MAX_HEALTH 500
+```
+
+This, however, performs a side effect of updating the health:
+```dm
+#define MOTH_SET_HEALTH(moth, new_health) ##moth.set_health(##new_health)
+```
+
+Now that you're caught up on the terms, let's get into the guidelines.
+
+### Put macro segments inside parentheses where possible.
+This will save you from bugs down the line with operator precedence.
+
+For example, the following macro:
+
+```dm
+#define MINIMUM_TEMPERATURE_FOR_SUPERCONDUCTION T20C + 10
+```
+
+...will break when order of operations comes into play:
+
+```dm
+var/temperature = MINIMUM_TEMPERATURE_FOR_SUPERCONDUCTION * 50
+
+// ...is preprocessed as...
+var/temperature = T20C + 10 * 50 // Oh no! T20C + 500!
+```
+
+This is [a real bug that tends to come up](https://github.com/tgstation/tgstation/pull/37116), so to prevent it, we defensively wrap macro bodies with parentheses where possible.
+
+```dm
+// Phew!
+#define MINIMUM_TEMPERATURE_FOR_SUPERCONDUCTION (T20C + 10)
+```
+
+The same goes for arguments passed to a macro...
+
+```
+// Guarantee 
+#define CALCULATE_TEMPERATURE(base) (T20C + (##base))
+```
+
+### Be hygienic where reasonably possible
+
+Consider the previously mentioned non-hygienic macro:
+
+```dm
+#define GET_HEALTH(health_percent) (##health_percent * max_health)
+```
+
+This relies on "max_health", but it is not obviously clear what the source is. This will also become worse if we *do* want to change where we get the source from. This would be preferential as:
+
+```dm
+#define GET_HEALTH(source, health_percent) (##health_percent * ##source.max_health)
```

Comment:
Should all of the "good" examples be following the other rules too? Will that make things more confusing or less confusing?

e.g. 
```suggestion
#define GET_HEALTH(source, health_percent) ((##health_percent) * (##source).max_health)
```